pkg/process/checks/allprocesses_windows.go

```diff
@@ -28,6 +29,7 @@ var (
 
 	// XXX: Cross-check state is stored globally so the checks are not thread-safe.
 	cachedProcesses = map[uint32]*cachedProcess{}
+	// TODO: remove locking here
```

Comment:
With these changes we no longer have to guard cachedProcesses with a lock because we are using the same goroutine to run both regular and RT checks (there is no contention).